File: pset1/cash/cash.c
#include <stdio.h>
#include <cs50.h>
#include <math.h>

int main(void)
{
    float change;
    // prompt user for change amount
    do
    {
        change = get_float("Change owed: ");
    }
    while (change < 0);

    int nCoins = 0;
    int cents = round(change * 100);
    // quarters
    int nQuarters = cents / 25;
    nCoins += nQuarters;
    int rest = cents % 25;
    // tens
    int nTens = rest / 10;
    nCoins += nTens;
    rest = rest % 10;
    // fives
    int nFives = rest / 5;
    nCoins += nFives;
    // the ones are the remainder of the division by 5
    nCoins += rest % 5;

    printf("%i\n", nCoins);
}

File: pset6/cash/cash.py
from cs50 import get_float
from math import floor


def main():
    change = get_change()
    nCoins = 0
    cents = round(100 * change)

    nQuarters = floor(cents / 25)
    nCoins += nQuarters
    rest = cents % 25

    nTens = floor(rest / 10)
    nCoins += nTens
    rest = rest % 10

    nFives = floor(rest / 5)
    nCoins += nFives

    # The ones are the remainder of the division by 5
    nCoins += rest % 5

    print(nCoins)


# Prompt user for change amount (positive)
def get_change():
    while True:
        f = get_float("Change owed: ")
        if f > 0:
            break
    return f


main()